src/storage/FileResourceStore.ts

```diff
@@ -94,42 +94,40 @@ export class FileResourceStore implements ResourceStore {
   public async deleteResource(identifier: ResourceIdentifier): Promise<void> {
     let path = this.parseIdentifier(identifier);
     if (path === '' || ensureTrailingSlash(path) === '/') {
-      throw new MethodNotAllowedHttpError('Cannot delete rootFilepath container.');
+      throw new MethodNotAllowedHttpError('Cannot delete root container.');
     }
-    path = `${this.rootFilepath}${path}`;
+    path = joinPath(this.rootFilepath, path);
+    let stats;
     try {
-      const stats = await fsPromises.lstat(path);
-      if (stats.isFile()) {
-        await fsPromises.unlink(path);
-        try {
-          await fsPromises.unlink(`${path}.metadata`);
-        } catch (_) {
-          // It's ok if there was no metadata file.
-        }
-      } else if (stats.isDirectory()) {
-        path = ensureTrailingSlash(path);
-        const files = await fsPromises.readdir(path);
-        let match = files.find((file): any => !file.startsWith('.metadata'));
-        if (match !== undefined) {
-          throw new ConflictHttpError('Container is not empty.');
-        }
+      stats = await fsPromises.lstat(path);
+    } catch (error) {
+      throw new NotFoundHttpError();
+    }
+    if (stats.isFile()) {
+      await fsPromises.unlink(path);
+      try {
+        await fsPromises.unlink(`${path}.metadata`);
+      } catch (_) {
+        // It's ok if there was no metadata file.
+      }
+    } else if (stats.isDirectory()) {
+      path = ensureTrailingSlash(path);
+      let files = await fsPromises.readdir(path);
+      let match = files.find((file): any => !file.startsWith('.metadata'));
+      if (match !== undefined) {
+        throw new ConflictHttpError('Container is not empty.');
+      }
 
+      match = files.find((file): any => file.startsWith('.metadata'));
+      while (match) {
```

Comment:
This store is indeed only going to create a .metadata file, but I left this loop untouched in case you want to go back to other and more `.`files later on. An update is then done faster by just modifying `'.metadata'`. It didn't seem to do much harm to leave this loop, but I can change it to a simple if-statement if you prefer?

Reply:
Yes I would just keep the `if`. If needed we can always add it later but for now it should only delete what it created itself.

Reply:
Comments though!